src/components/VariantMutationComparison/VariantMutationComparisonHook.ts

```diff
@@ -0,0 +1,85 @@
+import { pullAll } from '../../helpers/lodash_alternatives';
+import { MutationProportionEntry } from '../../data/MutationProportionEntry';
+import { ReferenceGenomeService } from '../../services/ReferenceGenomeService';
+import { LapisSelector } from '../../data/LapisSelector';
+import { useQuery } from '../../helpers/query-hook';
+import { MutationProportionData } from '../../data/MutationProportionDataset';
+import { useEffect, useMemo } from 'react';
+
+export function useOverlappingData(selectors: LapisSelector[], minProportion: number) {
+  const selectedVariants = useQuery(
+    signal => Promise.all(selectors.map(selector => MutationProportionData.fromApi(selector, 'aa', signal))),
+    [selectors],
+    useEffect
+  );
+  const overlappingData = useMemo(() => {
+    if (!selectedVariants || !selectedVariants.data || selectedVariants.data.length !== 2) {
+      return undefined;
+    }
+    const [mutationsFromVariant1, mutationsFromVariant2] = selectedVariants.data.map(v =>
+      selectMutationByProportion(v.payload, minProportion)
+    );
+    const { sharedMutations, mutationsOnlyInVariant1, mutationsOnlyInVariant2 } = findSharedMutations(
+      mutationsFromVariant1,
+      mutationsFromVariant2
+    );
+    const genes = groupByGenes(sharedMutations, mutationsOnlyInVariant1, mutationsOnlyInVariant2);
+    // Count
+    return [...genes.entries()].map(([gene, mutations]) => ({
+      gene,
+      ...mutations,
+    }));
+  }, [selectedVariants.data, minProportion]);
```

Comment:
IntelliJ: "ESLint: React Hook useMemo has a missing dependency: 'selectedVariants'. Either include it or remove the dependency array.(react-hooks/exhaustive-deps)"